python/tskit/trees.py

```diff
@@ -1205,6 +1205,33 @@ def right_sib_array(self):
         """
         return self._right_sib_array
 
+    def siblings(self, u):
+        """
+        Returns the sibling(s) of the specified node ``u`` as a tuple of integer
+        node IDs. If ``u`` has no siblings or is not a node in the current tree,
+        returns an empty tuple. If ``u`` is the root of a single-root tree,
+        returns an empty tuple; if ``u`` is the root of a multi-root tree,
+        returns the other roots (note all the roots are related by the virtual root).
+        If ``u`` is the virtual root (which has no siblings), returns an empty tuple.
+        If ``u`` is an isolated node, whether it has siblings or not depends on
+        whether it is a sample or non-sample node; if it is a sample node,
+        returns the root(s) of the tree, otherwise, returns an empty tuple.
+        The ordering of siblings  is arbitrary and should not be depended on;
+        see the :ref:`data model <sec_data_model_tree_structure>` section for details.
+
+        :param int u: The node of interest.
+        :return: The siblings of ``u``.
+        :rtype: tuple(int)
+        """
+        if u == self.virtual_root:
+            return tuple()
+        parent = self.parent(u)
+        if u in self.roots:
```

Comment:
I think this is more efficient as

```
if tree.num_samples(u) > 0 and tree.parent(u) == tskit.NULL
```

see #2620

Reply:
Or only checking once:

```
parent = self.parent(u)
if parent == tskit.NULL:
    if self.num_samples(u) == 0:
       return tuple()  # no parent and not a root
    else:
        parent = self.virtual_root
return tuple(v for v in self.children(parent) if u != v)
```

Reply:
I went with `if self.is_root(u):` so that `root_threshold` is taken into account.